### PR TITLE
changing api DELETE operation to async

### PR DESCRIPTION
**Pull request by maciejtreder**
# ARM (Control Plane) API Specification Update Pull Request 

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

**Comment by tomkerkhove** [comment]
@maciejtreder can you update the PR title to a representative title please?

**Comment by solankisamir** [comment]
@maciejtreder can you take care of the errors in the validation

image

**Comment by visingla-ms** [comment]
> Hi @maciejtreder! The automation detected breaking changes in this pull request. As a result, it added the `BreakingChangeReviewRequired` label. You cannot proceed with merging this PR until you complete one of the following action items: **ACTION ITEM ALTERNATIVE A**: Fix the breaking change. Please consult the documentation provided in the relevant validation failures. **ACTION ITEM ALTERNATIVE B**: Request approval. Alternatively, if you cannot fix the breaking changes, then you can request an approval for them. Please follow the process described in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process). **ACTION ITEM ALTERNATIVE C**: Report false positive. If you think there are no breaking changes, i.e. the validation should pass yet it fails, then please explain why in a PR comment and @ the PR assignee.

Please fix the breaking changes or get approval from Breaking Changes Review Board first. Afterwards, remove ARMChangesRequested label for ARM to review your swagger PR.

**Comment by solankisamir** [comment]
Tagging @raych1 for Go, @qiaozha for JavaScript

**Review by solankisamir — APPROVED** [review]

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimapis.json
Comment (Alancere) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimapis.json:464]: please revert this change to avoid duplicate schema.

Comment (maciejtreder) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-05-01-preview/apimapis.json]: Reverted.

**Comment by mentat9** [comment]
@maciejtreder - Your PR needs to pass checks and address the CI-BreakingChange-Python-Track2 label before merge. Add the MergeRequested label back after addressing those items. If you get stuck, you can ask for help in the API Spec Review Teams channel: https://teams.microsoft.com/l/channel/19%3a0351f5f9404446e4b4fd4eaf2c27448d%40thread.skype/API%2520Spec%2520Review?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47.

**Pull request merged**
